### PR TITLE
LibLine: Support Ctrl-b/f and Ctrl-d

### PR DESCRIPTION
**Pull request by nico**
b/f go backward/forward by one character, d deletes one character.

Also do some minor preparatory changes to make it easier to support these shortcuts.

### CODE REVIEW

**Comment by nico** [comment]
I'm never quite sure if these things should be handled in the terminal emulator or the shell btw. Maybe the Right Fix is to instead have the shell send the corresponding ansi escape code for these keys?

File: Libraries/LibLine/Editor.cpp [hunk 1]
```diff
@@ -422,6 +422,69 @@ void Editor::handle_read_event()
     Utf8View input_view { StringView { m_incomplete_data.data(), valid_bytes } };
     size_t consumed_codepoints = 0;
 
+    enum Amount { kCharacter, kWord };
```
Comment (alimpfard) [inline Libraries/LibLine/Editor.cpp:425]: uhhh...why not just `Character` and `Word`?

Comment (nico) [reply Libraries/LibLine/Editor.cpp]: Err, sorry. Fixed.

**Comment by awesomekling** [comment]
Quite tasteful!

**Pull request merged**